lib/src/wallet/keys/unified.rs

```diff
@@ -0,0 +1,406 @@
+            transparent_child_keys: vec![],
+            addresses: vec![],
+            next_sapling_diversifier_index: DiversifierIndex::new(),
+            encrypted: false,
+            unlocked: true,
+        }
+    }
+
+    pub fn new_from_phrase(
+        config: &ZingoConfig,
+        seed_phrase: &Mnemonic,
+        position: u32,
+    ) -> Result<Self, String> {
+        // The seed bytes is the raw entropy. To pass it to HD wallet generation,
+        // we need to get the 64 byte bip39 entropy
+        let bip39_seed = seed_phrase.to_seed("");
+        Ok(Self::new_from_seed(config, &bip39_seed, position))
+    }
+
+    pub(crate) fn get_all_taddrs(&self, config: &ZingoConfig) -> HashSet<String> {
+        self.addresses
+            .iter()
+            .filter_map(|address| {
+                address.transparent().and_then(|transparent_receiver| {
+                    if let zcash_primitives::legacy::TransparentAddress::PublicKey(hash) =
+                        transparent_receiver
+                    {
+                        Some(super::ToBase58Check::to_base58check(
+                            hash.as_slice(),
+                            &config.base58_pubkey_address(),
+                            &[],
+                        ))
+                    } else {
+                        None
+                    }
+                })
+            })
+            .collect()
+    }
+
+    pub fn first_sapling_address(&self) -> &zcash_primitives::sapling::PaymentAddress {
+        // This index is dangerous, but all ways to instanciate a UnifiedSpendAuthority
+        // create it with a suitable first address
+        self.addresses()[0].sapling().unwrap()
+    }
+
+    pub fn encrypt(&mut self, _passwd: String) -> std::io::Result<()> {
+        todo!()
+    }
+    pub fn lock(&mut self) -> std::io::Result<()> {
+        todo!()
+    }
+    pub fn unlock(&mut self, _passwd: String) -> std::io::Result<()> {
+        todo!()
+    }
+    pub fn remove_encryption(&mut self, _passwd: String) -> std::io::Result<()> {
+        todo!()
+    }
+}
+impl ReadableWriteable<()> for UnifiedSpendAuthority {
+    const VERSION: u8 = 1;
+
+    fn read<R: Read>(mut reader: R, _input: ()) -> io::Result<Self> {
+        let _version = Self::get_version(&mut reader)?;
+        let mut orchard_key_bytes = [0; 32];
+        reader.read_exact(&mut orchard_key_bytes)?;
+        let orchard_key: orchard::keys::SpendingKey =
+            Option::from(orchard::keys::SpendingKey::from_bytes(orchard_key_bytes)).ok_or(
+                io::Error::new(io::ErrorKind::InvalidData, "Invalid orchard spending key"),
+            )?;
+        let sapling_key = zcash_primitives::zip32::ExtendedSpendingKey::read(&mut reader)?;
+        let transparent_parent_key =
+            super::extended_transparent::ExtendedPrivKey::read(&mut reader, ())?;
+        let receivers_selected =
+            Vector::read(&mut reader, |mut r| ReceiverSelection::read(&mut r, ()))?;
```

Comment:
I don't understand why a `UnifiedSpendAuthority` would need more than one `ReceiverSelection`.

Reply:
Is this to support more than one seed_or_key from which the wallet has derived addresses?

Reply:
This is to support diversified addresses.